### PR TITLE
Configurable ssl file directory

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Created the `ssl_file_directory` variable for making the directory to put all ssl files into

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Updated the `mtls-customcerts-rhel` scenario to test out updating of the ssl_file_dir variable, both with/without trailing /

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.control_center/defaults/main.yml [hunk 1]
```diff
@@ -48,4 +48,4 @@ control_center_service_unit_overrides:
 ### Time in seconds to wait before starting Control Center Health Checks.
 control_center_health_check_delay: 30
 
-control_center_secrets_protection_file: /var/ssl/private/control-center-security.properties
+control_center_secrets_protection_file: "{{ ssl_file_directory }}control-center-security.properties"
```
Comment (erikgb) [inline roles/confluent.control_center/defaults/main.yml:51]:
```suggestion
control_center_secrets_protection_file: ""{{ (ssl_file_directory, 'control-center-security.properties') | path_join }}"
```

To avoid the requirement ^^ (MUST end in a backslash)

Comment (domenicbove) [reply roles/confluent.control_center/defaults/main.yml:51]: yes that is way better! 
But it says "New in version 2.10." ... i think we still have some users on 2.7, even though they are not supposed to

I don't think we could ask anyone to update ansible for a 6.1.x change, but maybe with 6.2.0

I suppose well be asking ppl to change to 2.11 soon though...

Comment (domenicbove) [reply roles/confluent.control_center/defaults/main.yml:51]: another option... i've been doing this elsewhere is to strip the last slash. I could strip it w regex and then add it back.
```
_ssl_file_directory: "{{ ssl_file_directory |regex_replace('\\/$', '') }}/"
```

Comment (domenicbove) [reply roles/confluent.control_center/defaults/main.yml]: I went w the second approach, just so this change would not require users to do anything (upgrade ansible). Good suggestion on removing that requirement. LMK what you think

File: VARIABLES.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ Below are the supported variables for the role confluent.variables
 
 ***
 
+### confluent_package_version
```
Comment (erikgb) [inline VARIABLES.md:7]: I do not understand why this is added in this PR? Leftovers from something else?

File: roles/confluent.variables/defaults/main.yml [hunk 3]
```diff
@@ -136,6 +142,12 @@ ssl_self_signed_ca_cert_filepath: generated_ssl_files/snakeoil-ca-1.crt
 ssl_self_signed_ca_key_filepath: generated_ssl_files/snakeoil-ca-1.key
 ssl_self_signed_ca_password: capassword123
 
+### Directory on hosts to store all ssl files.
+ssl_file_dir: /var/ssl/private/
+
+# Confirm a trailing / on the ssl file directory path
+ssl_file_dir_final: "{{ ssl_file_dir |regex_replace('\\/$', '') }}/"
```
Comment (erikgb) [inline roles/confluent.variables/defaults/main.yml:149]:
```suggestion
ssl_file_dir_final: "{{ ssl_file_dir | regex_replace('\\/$', '') }}/"
```

File: roles/confluent.variables/defaults/main.yml [hunk 4]
```diff
@@ -1,8 +1,14 @@
 ---
 # Custom filters used in this file are defined in filter_plugins/filters.py
 
-confluent_repo_version: 6.1
+confluent_repo_version: "{{ confluent_package_version | regex_replace('^([0-9])\\.([0-9]*).*', '\\1.\\2') }}"
```
Comment (erikgb) [inline roles/confluent.variables/defaults/main.yml:4]: This is an unrelated change to this PR. I would suggest dropping this change here, and I would also question this change in general. 😉 I think it is nice to see the repo-version in "clear text" - even if semi-duplicated....

**Review by erikgb — CHANGES_REQUESTED** [review]
My personal preference would be to remove the trailing `/` from the `ssl_file_dir_final` variable, and add it whereever is is used. More text, but I think it is easier to read and more intuitive.

File: roles/confluent.variables/defaults/main.yml (hunk 4 above)
Comment (domenicbove) [reply roles/confluent.variables/defaults/main.yml:4]: ya the point is that why would you need to set 
```
confluent_package_version: 6.1.1
confluent_repo_version: 6.1
```
when just `confluent_package_version` is enough. All in all `confluent_package_version` is not in VARIABLES.md and it should be

Comment (erikgb) [reply roles/confluent.variables/defaults/main.yml:4]: I do not necessarily disagree with the change. 😄 But I think it could be a separate PR, since I prefer PRs with a single concern. And I also think the variable `confluent_repo_version` should be moved to vars (not defaults) - since it is now a deduced variable.

File: VARIABLES.md (hunk 2 above)
Comment (domenicbove) [reply VARIABLES.md:7]: I want to "sneak this in" bc its missing. Do you feel this really needs a separate PR?

File: roles/confluent.variables/defaults/main.yml (hunk 4 above)
Comment (domenicbove) [reply roles/confluent.variables/defaults/main.yml]: ok lets just do 2 prs, thats not that hard

File: roles/confluent.ssl/tasks/import_ca_chain.yml [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
   file:
-    path: /var/ssl/private/generation/trustCAs
+    path: "{{ ssl_file_dir_final }}/generation/trustCAs"
     state: directory
     mode: 0755
 
 - name: Split CA Certificate Bundle into Cert Files
   shell: |
-    cat {{ca_cert_path}} | awk 'split_after==1{n++;split_after=0} /-----END CERTIFICATE-----/ {split_after=1} {print > ("/var/ssl/private/generation/trustCAs/ca" n ".pem")}'
+    cat {{ca_cert_path}} | awk 'split_after==1{n++;split_after=0} /-----END CERTIFICATE-----/ {split_after=1} {print > ("{{ ssl_file_dir_final }}generation/trustCAs/ca" n ".pem")}'
```
Comment (erikgb) [inline roles/confluent.ssl/tasks/import_ca_chain.yml:10]:
```suggestion
    cat {{ca_cert_path}} | awk 'split_after==1{n++;split_after=0} /-----END CERTIFICATE-----/ {split_after=1} {print > ("{{ ssl_file_dir_final }}/generation/trustCAs/ca" n ".pem")}'
```

File: roles/confluent.variables/defaults/main.yml [hunk 6]
```diff
@@ -3,6 +3,7 @@
 
 confluent_repo_version: 6.1
 confluent_package_version: 6.1.1
+
```
Comment (erikgb) [inline roles/confluent.variables/defaults/main.yml:6]:
```suggestion
```

**Review by erikgb — CHANGES_REQUESTED** [review]
Two minor comments, close to LGTM....

**Review by erikgb — APPROVED** [review]
LGTM, but I might have overlooked something. 😅 Major changes. Great job, @domenicbove!

**Comment by domenicbove** [comment]
@erikgb Thanks for the review! I thought I did a complete scan but you found a missing slash! I'll take another look
